eng/testing/tests.targets

```diff
@@ -150,7 +150,7 @@
         Abi="$(AndroidAbi)"
         ProjectName="$(AssemblyName)"
         MonoRuntimeHeaders="$(NETCoreAppRuntimePackNativePath)\include\mono-2.0"
-        MainLibraryFileName="AndroidTestRunner.dll"
+        MainLibraryFileName="$(AndroidTestRunner)\$(HostArch)\$(Configuration)\AndroidTestRunner.dll"
```

Comment:
Are you sure in this? I see https://github.com/dotnet/runtime/blob/0fa0b905f65e57f5eb64dd556a07ead522930c8a/src/mono/msbuild/AndroidAppBuilder/ApkBuilder.cs#L30-L31 that the file is expected to be in the app dir so a file name is expected instead of full path

Reply:
I guess it's AppleAppBuilder-only issue (after @akoeplinger's changes?)

Reply:
This [PR](https://github.com/dotnet/runtime/pulls?q=is%3Apr+is%3Aclosed+author%3Amdh1418) moved both the TestRunner and AppBuilder binaries for both iOS and Android into the `artifacts/bin` folder. 
After building, I am getting this
```
~/runtime$ls artifacts/bin/AppleAppBuilder/x64/Release/
AppleAppBuilder.deps.json	AppleAppBuilder.dll
```
Also running the command `./build.sh -c Release -os iOS -subset Mono+Libs+Libs.Tests` with these changes led to a successful build

Reply:
Um.. can you please check that it actually runs the tests? Because it can build but simply crash on start now.

Reply:
yeah this isn't the right change, it'd need to modify `AppleTestRunnerBinaries` which is already done by https://github.com/dotnet/runtime/pull/36157.